### ISSUE TITLE
guitar sniffer 1.1 stops working

### ISSUE THREAD

**Issue by toweltan** [artman41/guitarsniffer#8]
ran guitar sniffer for a few days everything goes well then it suddenly stops detecting my button inputs. i changed nothing but the old sniffer works fine. are other people experiencing this problem where is just stops working randomly?

**Comment by krpotter** [comment]
Have similar issue. Old sniffer works fine, but this new sniffer won't detect any keys at all.  Added my details to the end of issue #6, which is still unresponded to. Not sure if the author is maintaining this anymore or not.

**Comment by Dunkalunk** [comment]
I've been in contact with artman as I've been taking what he built for the guitar and making it work for the RB4 drums.  He's been busy with life so I'm not sure how much he's paying attention here.

Anyway, I've learned a bit about how this stuff works but not sure how good I'll be at troubleshooting.  I've also not experienced the issue and I've been using the sniffer without issues for months now.

Where's the link to the previous sniffer program?  I'll try to see what the difference might be and why it's not working for you guys.

I'm wondering if antivirus is blocking something with any of the packet programs or the sniffer itself?  Can you guys check your antivirus or Windows protection and see if any of the different programs used are blocked at all?  Not sure why they would be but who knows.

@krpotter I noticed you did quite a bit of troubleshooting yourself looking into the controller actually connecting to vJoy, so I'm not sure what else might be wrong, but maybe we'll find something together.

**Comment by krpotter** [comment]
Artman's original sniffer was called FenderStratocastorSniffer, and is here:
  https://github.com/artman41/FenderStratocastorSniffer

The very latest version of that is 4.0, but I, like several others I read about, couldn't get v4.0 to work.  Instead we had to use the version 3.0 that he has posted on that page, which does work.  

That older project (FenderStratocastorSniffer) is apparently dead now, and has been replaced by Artman with his newer GuitarSniffer, but again I and several others I've seen can't seem to get this newer version to work on Windows 10 (but then again some folks have no issues getting it to work).  I've completely disabled A/V and Windows Defender, even the Windows Firwall, to try to get it to work, but to no avail.  Unfortunately, no keystrokes at all show up in the GuitarSniffer GUI.  It's got to be either an issue with the vJOY program, or the GuitarSnifferGUI, as I know packets are getting through from the Guitar through WinPCAP and USBPCap, as both of these are used by the older FenderStratocasterSniffer, and that works fine (at least version 3.0 of that). 

Thanks for any light you can shed.

**Comment by toweltan** [comment]
So I've figured out a way to consistently get giu sniffer to work on win10. It works as soon as I restart my pc. If my pc has been on for a while it doesnt work. So whenever I want to play CH I restart my pc and open giu sniffer then it works.

**Comment by Dunkalunk** [comment]
That's weird...  What are the specs on your PC?  Just want to see what you're working with.  You can send a screenshot of System from the Control Panel.  
Just a reminder that this is unofficial help.  I'm not the writer of the program, I've just been working on it to get it working for drums haha so I'm just going to try my best to help.

**Comment by krpotter** [comment]
That's strange that a reboot would cause things to start working for toweltan.  Sounds like a system resource (RAM?) is getting used up, or something else is getting started at some point after the reboot, that is interfering with the GuitarSniffer program.  I've tried running GuitarSnifferGUI right after rebooting my PC, and still no luck.  As an FYI, my system config is as follows.  A fairly high end machine, as I run the Oculus VR system on it, as well as other software development:

System:           Windows 10 Home
Processor:       Intel Core i7-6700 CPU @ 3.40GHZ
Install RAM:    16.0 GB Dual-Channel @ 1066Mhz
System Type:   64-Bit Operating System, x64-based processor
Motherboard:  MSI Z170I Gaming Pro AC (MS-7980)
Graphics:         Nvidia GeForce GTX 970 (MSI)
HardDrives:     Dual SSD drives RAIDed (Mirroring)

I'm also a Software developer/programmer, just have never coded in Google's GO language (have lots of experience in C, C++, VB, VB.net, Perl, etc).  Since it's very similar to "C", I might have to have a go at it, and see if I can add some debugging lines into the guitarjoypad.go, guitarpacket.go and guitarsniffer.go, to see what I can find for my particular situation.  Dunkalunk, thanks for your help on the issues.  Totally understand this is all free help for one another :-)

**Comment by Dunkalunk** [comment]
@krpotter yeah that should be more than enough.  There is another issue open for how much CPU it uses, but I'm not sure if Artman has looked into that at all.  But when I first started using it and developing the drum sniffer I was on an i7-7700HQ in a laptop with 16 GB of RAM and a 1050Ti, so what you've got should be more than enough.  If @toweltan is anywhere close to those specs then it must be something else.  
But still, they're two separate issues.  Yours won't work at all, Toweltan's just dies eventually and a restart fixes it.  Weird.

You've probably checked this since you've been able to do a bunch of troubleshooting already, but when GuitarSniffer is running can you open up the vJoy Device List and make sure the controller's ID is listed there?

**Comment by krpotter** [comment]
Yes indeed, the device ID is listed in vJoy when I start up guitar sniffer.  If I close guitar sniffer, the ID then goes away. It shows up as the first device in the vJoy list.

**Comment by ghost** [comment]
This has stopped working for me today.

Has worked fine with no issues for literally over a month and all of a sudden the sniffer isn't detecting any input.

Reallly sad about it :(

**Comment by Rhystic1** [comment]
Same issue - working perfectly for months, then suddenly detects no inputs.

Sometimes this would happen but then after rebooting it would resume working, but this time even after reinstalling everything and rebooting this is still happening. Very annoying :(

**Comment by Dunkalunk** [comment]
@Rhystic1 and anyone else that ends up having this issue, I recommend checking out [this software here](https://github.com/TheNathannator/RB4InstrumentMapper) for RB4 instruments.  It's the latest tool recommended by the Clone Hero team and is currently receiving regular updates.  I haven't tested it on drums yet but can definitely recommend it for guitar!
